SQL/SQL.sublime-syntax

```diff
@@ -249,7 +251,52 @@ contexts:
       # Handle situations where the schema and . 
     - match: '{{end_identifier}}'
       pop: true
-  like_escape:
+  like_string_not_followed_by_escape:
+    - include: comments
+    - match: \'
+      scope: punctuation.definition.string.begin.sql
+      set: [like_escape_fail, inside_like_single_quoted_string]
+    - match: (?=\S)
+      pop: true
+  like_escape_fail:
+    - include: comments
+    - match: (?i:\bescape\b)
+      fail: like_string_branch
+    - match: (?=\S)
+      pop: true
+  like_escape_pop:
+    - include: comments
+    - match: (?i:\bescape\b)
+      scope: keyword.operator.word.sql
+      pop: true
+    - match: (?=\S)
+      pop: true
```

Comment:
I'd move those contexts down to the other `like_escape_...` contexts to keep `like_string_...` contexts logically grouped together and sorted by "complexity".

A syntax should probably always start with some managing contexts containing only includes, followed by complex contexts representing statements  etc. and finalized by simple contexts (prototypes?) which are re-used often - just like a tree. The `like_escape_fail` are leaf contexts as they don't push others onto stack nor do they include any other contexts (if we ignore the comments ;-) )